File: templates/variants/bun-hono/src/temporal/activities.ts
export type WaitlistFollowUpInput = {
  triggerId?: string;
  email?: string;
  type: string;
};

export async function recordWaitlistFollowUp(input: WaitlistFollowUpInput) {
  return {
    status: "queued",
    triggerId: input.triggerId ?? null,
    email: input.email ?? null,
    type: input.type,
  };
}

File: templates/variants/go-connectrpc/internal/temporal/activities.go
package temporalapp

import "context"

type WaitlistFollowUpInput struct {
	TriggerID string
	Email     string
	Type      string
}

type WaitlistFollowUpResult struct {
	Status    string
	TriggerID string
	Email     string
	Type      string
}

type Activities struct{}

func (a *Activities) RecordWaitlistFollowUp(ctx context.Context, input WaitlistFollowUpInput) (WaitlistFollowUpResult, error) {
	return WaitlistFollowUpResult{
		Status:    "queued",
		TriggerID: input.TriggerID,
		Email:     input.Email,
		Type:      input.Type,
	}, nil
}
